test/e2e/test/elasticsearch/steps_mutation.go

```diff
@@ -192,8 +194,16 @@ func (hc *ContinuousHealthCheck) Start() {
 				// recreate the Elasticsearch client at each iteration, since we may have switched protocol from http to https during the mutation
 				client, err := hc.esClientFactory()
 				if err != nil {
-					// treat client creation failure same as unavailable cluster
-					hc.AppendErr(err)
+					// according to https://github.com/kubernetes/client-go/blob/fb61a7c88cb9f599363919a34b7c54a605455ffc/rest/request.go#L959-L960,
+					// client-go requests may return *errors.StatusError or *errors.UnexpectedObjectError, or http client errors.
+					switch err.(type) {
+					case *k8serrors.StatusError, *k8serrors.UnexpectedObjectError:
+						// explicit apiserver error, consider as healthcheck failure
+						hc.AppendErr(err)
+					default:
+						// likely a network error, log and ignore
+						fmt.Printf("error while creating the Elasticsearch client: %s", err.Error())
```

Comment:
nit: I _think_ `.Error()` is not required here, according to https://golang.org/pkg/fmt/

> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).